src/CMakeLists.txt

```diff
@@ -177,8 +167,42 @@ set(IIDM_SOURCES
     xml/XmlStreamWriter.cpp
 )
 
-add_library(iidm ${IIDM_SOURCES})
+# Shared library
+add_library(iidm SHARED ${IIDM_SOURCES})
+set_target_properties(iidm PROPERTIES SOVERSION ${IIDM_SOVERSION} VERSION ${IIDM_VERSION})
+target_include_directories(iidm SYSTEM PUBLIC ${Boost_INCLUDE_DIRS} ${LIBXML2_INCLUDE_DIR})
 target_include_directories(iidm
-    PRIVATE ${CMAKE_HOME_DIRECTORY}/include
-    PRIVATE ${CMAKE_HOME_DIRECTORY}/src/
+    INTERFACE
+        $<INSTALL_INTERFACE:${INSTALL_INCLUDE_DIR}>
+    PUBLIC
+        $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/include>
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
+)
+target_link_libraries(iidm ${LIBXML2_LIBRARIES})
```

Comment:
I would link to the target if find_package(libxml2) creates one. And then target_include_directories may not be required if the target is well formed. I would add PRIVATE too.

Reply:
See below: I disagree about the PRIVATE. LibXML2 is exposed in the interface so it should not be private, right?